### ISSUE TITLE
Delete vehicles from the game

### ISSUE THREAD

**Issue by deferred-impact** [deferred-impact/battledudes-issue-tracker#206]
Vehicles make this game worse. They introduce gameplay depth, sure; but the gameplay they add is not good, nor skillful, nor fair. Nothing is fair about being insta-flattened by an ATV. Nothing is fair about being blasted by a tank you couldn't see because it has a longer vision range than you.

B-but we need vehicles for traveling across maps! No we don't. Maps do not need to be this huge in the first place. It's an invented problem, to which vehicles are your solution. Simply reduce the map size and vehicles are irrelevant. Compound exists just fine without them.

Or if you absolutely must have large maps, then offer players a different way for moving around. I'd simply make it so that a player can pick a location on the map and, after staying stationary for 5 seconds (if they move it is cancelled) and glowing, they straight up teleport to the point they picked.

Depending on the chosen implementation the effort will vary. I am tagging this as high effort assuming all large maps will need to be redesigned and reduced in size.

**Comment by ghost** [comment]
Nah, removing vehicles wouldn't be the best idea. I do agree however, they must be nerfed seeing their instant DPS.

**Comment by ghost** [comment]
What you are talking about above, would be a teleporter of some kind, that allows a player to spawn randomly on the map. 

A. That would be ridiculously overpowered, they could spawn in the best possible locations, which would give an unfair disadvantage to their opponent. Imagine you have the flag, in CTF, Upload, or Football, and a random player teleports next to you, and shoots you down almost instantly before you could even react. 

B. Fights will never end. Normally, with a vehicle, there is a certain amount of time delay for one to reach from point A to point B, because cars are only so fast, and the driving skills of a player are only so precise. However, with a teleporter, all you need is a 10 second cooldown, and BAM. You'll be there to fight. In the situation of a flag gamemode, this is ridiculous, because the flag holder will have no opportunity to carry the flag safely, they'll be ambushed almost infinitely. 

C. Teleportation accuracy. What if the player is teleported to the exact same location as an object, or player? If inside an object, they will be stuck, and might reach a weird location because of the physics engine. If they crash into a person, a moving target, both could end up in a weird location, but most importantly, one player will be able to catch the other by surprise, because neither the teleporter or the enemy will expect to end up in such a situation (Assuming that you are kept from seeing people on the map when you try to teleport.

**Comment by ghost** [comment]
How about, instead of reducing map size, we reduce **car damage**, and **how the damage is given.** Normally, when someone is driven over by a car, they take damage for as long as they remain in contact with the vehicle, and since vehicles have high speed, they instantly kill a player if aimed correctly and and fast enough. 
However, if vehicles did damage per amount of times a player comes into contact, rather than doing damage per the amount of time a player _remains_ in contact, then killing with vehicles won't be as overpowered. Once a car hits an enemy, they take a specific amount of damage, but then that's it. They don't take anymore, because they remain in contact, thus preventing them from dying instantly. it would be like hitting barbed wire, you only take damage once, when you come into contact, but not for remaining in contact. In order to take damage from it again, you'd have to lose physical contact of the vehicle, and then touch it again.

**Comment by ghost** [comment]
Please ignore my comments. This was already suggested by Hoax.

**Comment by deferred-impact** [comment]
> Nah, removing vehicles wouldn't be the best idea. I do agree however, they must be nerfed seeing their instant DPS.

I think it would be wonderful. This is a shooter game, not a racing game, it should focus on the shooting. Many maps already ban vehicles, and work just fine.

> That would be ridiculously overpowered, they could spawn in the best possible locations, which would give an unfair disadvantage to their opponent. Imagine you have the flag, in CTF, Upload, or Football, and a random player teleports next to you, and shoots you down almost instantly before you could even react.

This can be balanced properly. Add a delay for this teleportation. During this delay you're stuck at your destination, unable to shoot, and unable to take damage, and glowing in some way i guess. Give it 5 seconds or so. This would remove the ability to surprise attack opponents via the teleport.

> Fights will never end. Normally, with a vehicle, there is a certain amount of time delay for one to reach from point A to point B, because cars are only so fast, and the driving skills of a player are only so precise. However, with a teleporter, all you need is a 10 second cooldown, and BAM. You'll be there to fight. In the situation of a flag gamemode, this is ridiculous, because the flag holder will have no opportunity to carry the flag safely, they'll be ambushed almost infinitely.

The same "certain amount of time delay" that exists with a car, could be implemented with a teleport. Make it increase the further you are teleporting.

> Teleportation accuracy. What if the player is teleported to the exact same location as an object, or player? If inside an object, they will be stuck, and might reach a weird location because of the physics engine. If they crash into a person, a moving target, both could end up in a weird location, but most importantly, one player will be able to catch the other by surprise, because neither the teleporter or the enemy will expect to end up in such a situation (Assuming that you are kept from seeing people on the map when you try to teleport.

A simple algorithm that seeks out a nearby "safe destination" would fix this.

> Please ignore my comments. This was already suggested by Hoax.

That would be #214 , i believe

> ...

I will grant you one thing about the teleporter idea - we can discuss it for a long time, and it's an interesting discussion, but there is absolutely no way in hell the developer of this game actually implements it. It's too far-fetched for his "my game must copy call of tooty" mantra.

**Comment by ghost** [comment]
> I think it would be wonderful. This is a shooter game, not a racing game, it should focus on the shooting. Many maps already ban vehicles, and work just fine.

Fair enough. 

>  Give it 5 seconds or so. This would remove the ability to surprise attack opponents via the teleport.

How, exactly? Even with a 5 sec delay, you can still surprise someone, determining that a player will travel from the point they were at drastically, is quite an assumption. 

> A simple algorithm that seeks out a nearby "safe destination" would fix this.

Oh? Perfect. It will be just like the in game spawn system, where you can spawn inside walls and objects, perfect for 'safety spawn'. 

> That would be https://github.com/deferred-impact/battledudes-issue-tracker/issues/214 , i believe

Yeah. 

> but there is absolutely no way in hell the developer of this game actually implements it. It's too far-fetched for his "my game must copy call of tooty" mantra.

Agreed. Doesn't sound like he'll even consider it, seeing his previous contributions.

**Comment by deferred-impact** [comment]
> determining that a player will travel from the point they were at drastically, is quite an assumption.

Wha?

> Oh? Perfect. It will be just like the in game spawn system, where you can spawn inside walls and objects, perfect for 'safety spawn'.

My ideas do hinge on the idea of an intelligent developer implementing them. Not my fault the dev of this game can't get anything right, lmao.

> Agreed. Doesn't sound like he'll even consider it, seeing his previous contributions.

A lot of this github is suggesting for suggesting's sake. Fun to ponder about game design using an existing game as a source of concrete examples.

**Comment by ghost** [comment]
> Wha?

Even with a 5 second cooldown, you could jumpscare a player by teleporting next to them, because you can't be sure that a 5 second delay will ensure that the targeted player will move. They could remain in the same spot, and an enemy just has to teleport next to them, and will kill them by surprise, whether there is a 5 second delay or not.

**Comment by deferred-impact** [comment]
> Even with a 5 second cooldown, you could jumpscare a player by teleporting next to them, because you can't be sure that a 5 second delay will ensure that the targeted player will move. They could remain in the same spot, and an enemy just has to teleport next to them, and will kill them by surprise, whether there is a 5 second delay or not.

The player porting in would be visible during the 5 seconds, but invincible. Perhaps they would be able to move. They wouldn't be able to deal any damage during the 5 seconds. No surprise attack possible.

**Comment by ghost** [comment]
> The player porting in would be visible during the 5 seconds, but invincible. Perhaps they would be able to move. They wouldn't be able to deal any damage during the 5 seconds. No surprise attack possible.

Spawn protection? Sure, works for me.
